functions/typed/Greeting/Function.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START functions_typed_greeting]
+using Google.Cloud.Functions.Framework;
+using System.Text.Json.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Greeting;
+
+public class TypedFunction : ITypedFunction<GreetingRequest, GreetingResponse>
```

Comment:
I'm not sure whether this has changed or I just didn't spot it before. Our templates all create functions just called Function, and that's what all the other samples use. If you really, really want to use the class name TypedFunction, please rename the file as well - but I think it would be better to reset the name to Function.